src/main/java/dk/kb/util/webservice/OpenApiResource.java

```diff
@@ -0,0 +1,185 @@
+package dk.kb.util.webservice;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import dk.kb.util.Resolver;
+import dk.kb.util.string.CallbackReplacer;
+import dk.kb.util.webservice.exception.NotFoundServiceException;
+import dk.kb.util.yaml.YAML;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ * Handle serving of OpenAPI specification for a webapp. This class handles dynamic updates of the API specification.
+ * Through this class it gets possible to use syntax as the following {@code ${config:yaml.path}} to access values from
+ * config files in API specifications.
+ */
+public class OpenApiResource extends ImplBase {
+    private static final Logger log = LoggerFactory.getLogger(OpenApiResource.class);
+
+    /**
+     * The config where values are substituted from.
+     */
+    static private YAML config;
+
+    public static final String APPLICATION_YAML = "application/yaml";
+
+    /**
+     * Pattern to allow search-replace for variabels defined as ${config.yaml.path} in OpenAPI specifications.
+     * Everything after 'config.' is treated as a path to an entry in the backing configuration.
+     */
+    private static final Pattern CONFIG_REPLACEMENT= Pattern.compile("\\$\\{config:([^}]+)}");
+
+    /**
+     * Replacer that use {@link #CONFIG_REPLACEMENT} for matching and {@link #getReplacementForMatch(String)}
+     * for getting the replacement.
+     */
+    private static final CallbackReplacer CONFIG_PROCESSOR = new CallbackReplacer(
+            CONFIG_REPLACEMENT, OpenApiResource::getReplacementForMatch, true);
+
+
+    /**
+     * JAX-RS needs the empty constructor for serving the webapp. To configure the {@link OpenApiResource} create an
+     * instance of the class using the {@link OpenApiResource(YAML)} constructor inside the given implementation of
+     * {@link Application#getClasses()} before returning all classes that are part of the application.
```

Comment:
This is the primary way of using the class. Could you repeat the instructions in the Javadoc for the class? Preferably with a code sample of the `Application`-code?